core-bundle/src/Controller/Base64EncodedRedirectController.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller;
+
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\UriSigner;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @internal
+ */
+class Base64EncodedRedirectController
+{
+    /**
+     * @var UriSigner
+     */
+    private $uriSigner;
+
+    public function __construct(UriSigner $uriSigner)
+    {
+        $this->uriSigner = $uriSigner;
+    }
+
+    /**
+     * @Route("/_contao/base64_redirect", name="contao_base64_redirect")
+     */
+    public function renderAction(Request $request): Response
+    {
+        if (!$request->query->has('redirect')) {
+            throw new BadRequestHttpException();
+        }
+
+        // We cannot use $request->getUri() here as we want to work with the original URI (no query string reordering)
+        if (!$this->uriSigner->check($request->getSchemeAndHttpHost().$request->getBaseUrl().$request->getPathInfo().(null !== ($qs = $request->server->get('QUERY_STRING')) ? '?'.$qs : ''))) {
```

Comment:
Could we have something a bit more readable here? :-)
```suggestion
        $uri = $request->getSchemeAndHttpHost() . $request->getBaseUrl() . $request->getPathInfo() .
          (null !== ($qs = $request->server->get('QUERY_STRING')) ? '?'.$qs : '');
          
        if (!$this->uriSigner->check($uri)) {
```

Reply:
Will come with the next Symfony version https://github.com/symfony/symfony/pull/35284 :D

Reply:
btw.: Would this be needed in the future anywhere again? Can't we have a a request signer or sth. to wrap this duplicated concat logic anywhere?

Reply:
It's not duplicated and it's already fixed in Symfony 5.1.
Can we proceed to merging this finally now?

Reply:
Ah, yes. I just discovered https://github.com/symfony/symfony/pull/35284 :D
Fine with me!

Reply:
@richardhj You comment did not show up for me until @Toflar s was displayed. So sorry for the confusion.